src/plugins/usage_collection/server/usage_counters/usage_counter.ts

```diff
@@ -20,6 +21,12 @@ export interface UsageCounterParams {
  * API whenever the event happens.
  */
 export interface IUsageCounter {
+  /**
+   * Defines custom retention period for the counters under this domain.
+   * This is the number of days worth of counters that must be kept in the system indices.
+   * Defaults to 5
```

Comment:
nit: I don't see us defaulting it anywhere here.

How about defaulting `retentionPeriodDays = USAGE_COUNTERS_KEEP_DOCS_FOR_DAYS` if not provided?